In [1]:
#Mengkoneksikan Drive pada Colab

from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [2]:
import pandas as pd
import numpy as np
import os
import glob

In [3]:
#Pilih direktori Tokped
os.chdir("/content/drive/My Drive/Dataset/Laptop_Price_List_Tokopedia/Tokped")

In [9]:
df = pd.read_csv('tokped_clean_data.csv')
df

name  ...        date
0       ACER ASPIRE 3 A314 - INTEL CELERON N4000 - 14 ...  ...  2020-10-02
1       Laptop Acer Aspire 3 Celeron N4120 256GB SSD 4...  ...  2020-10-02
2       LAPTOP ACER ASPIRE 3 A314-21 A4 9120e RAM 4GB ...  ...  2020-10-02
3       Acer Aspire 3 A314 A4 9120 4GB 1TB W10 14.0 DV...  ...  2020-10-02
4       PC All In One acer Aspire C24-960 8GB 1TB Core...  ...  2020-10-02
...                                                   ...  ...         ...
150683  ASUS VIVOBOOK FLIP TM420IA RYZEN 5-4500U 8GB 5...  ...  2021-01-26
150684  Laptop Asus Vivobook K413FQ i5 8gb 512Gb 14fhd...  ...  2021-01-26
150685  Asus vivobook A416ma intel N4020 4gb 256gb 14 ...  ...  2021-01-26
150686  ASUS VIVOBOOK A407UF I5 8250U 4GB 256GB MX130 ...  ...  2021-01-26
150687  LAPTOP ASUS VIVOBOOK K413EQ I7-1165G7 8GB 512G...  ...  2021-01-26

[150688 rows x 6 columns]

## Pembersihan Data
Membersihkan data dari data yang tidak digunakan. Kolom `link` tidak akan digunakan sehingga dibuang saja.

In [10]:
#membuang kolom 'link'
df.drop('link', axis=1, inplace=True)
df

name  ...        date
0       ACER ASPIRE 3 A314 - INTEL CELERON N4000 - 14 ...  ...  2020-10-02
1       Laptop Acer Aspire 3 Celeron N4120 256GB SSD 4...  ...  2020-10-02
2       LAPTOP ACER ASPIRE 3 A314-21 A4 9120e RAM 4GB ...  ...  2020-10-02
3       Acer Aspire 3 A314 A4 9120 4GB 1TB W10 14.0 DV...  ...  2020-10-02
4       PC All In One acer Aspire C24-960 8GB 1TB Core...  ...  2020-10-02
...                                                   ...  ...         ...
150683  ASUS VIVOBOOK FLIP TM420IA RYZEN 5-4500U 8GB 5...  ...  2021-01-26
150684  Laptop Asus Vivobook K413FQ i5 8gb 512Gb 14fhd...  ...  2021-01-26
150685  Asus vivobook A416ma intel N4020 4gb 256gb 14 ...  ...  2021-01-26
150686  ASUS VIVOBOOK A407UF I5 8250U 4GB 256GB MX130 ...  ...  2021-01-26
150687  LAPTOP ASUS VIVOBOOK K413EQ I7-1165G7 8GB 512G...  ...  2021-01-26

[150688 rows x 5 columns]

Dalam dataset juga terdapat data yang bukan laptop, sehingga perlu disaring lalu dibuang. Data yang mengandung string berikut pada kolom `name` perlu dibuang :

`['pc design','pc gaming','panasonic','compresor','ac daikin', 'ac gree', 'phone', 'asus rog 2', 'asus rog 3', 'rogphone', 'hand pallet', 'kyoritsu', 'power supply', 'daikin', 'yoke mpi','isolasi pipa','mini power']`

In [11]:
# periksa data yang bukan laptop untuk dibuang

searchfor = ['pc design','pc gaming','panasonic','compresor','ac daikin', 'ac gree', 'phone', 'asus rog 2', 'asus rog 3', 'rogphone', 'hand pallet', 'kyoritsu', 'power supply', 'daikin', 'yoke mpi','isolasi pipa','mini power']
df[df['name'].str.contains('|'.join(searchfor), case=False)]

name  ...        date
3277    ASUS ROG Phone 3 5G 865+ Global 12GB/128GB 256...  ...  2020-10-20
3279    Asus ROG Phone 3 8/128 Garansi RESMI SD 865+ R...  ...  2020-10-20
3341    ASUS ROG Phone 3 5G 865+ Global 12GB/128GB 256...  ...  2020-10-20
3343    Asus ROG Phone 3 8/128 Garansi RESMI SD 865+ R...  ...  2020-10-20
3395    ASUS ROG PHONE 2 8/128GB -12/512GB RAM INTERNA...  ...  2020-10-20
...                                                   ...  ...         ...
147591                                         Asus ROG 2  ...  2021-01-25
147593  Asus Rog Phone 2 8/128 Gb Garansi resmi indonesia  ...  2021-01-25
147594  Asus ROG Phone 3 RAM 8GB INTERNAL 128GB Snapdr...  ...  2021-01-25
147596         ASUS ROG 2 RAM 12/512 GB GARANSI RESMI TAM  ...  2021-01-25
147597                             asus rog phone 3 resmi  ...  2021-01-25

[4173 rows x 5 columns]

In [12]:
# lalu buang data-data tersebut

searchfor = ['pc design','pc gaming','panasonic','compresor','ac daikin', 'ac gree', 'phone', 'asus rog 2', 'asus rog 3', 'rogphone', 'hand pallet', 'kyoritsu', 'power supply', 'daikin', 'yoke mpi','isolasi pipa','mini power']
df2 = df[~df['name'].str.contains('|'.join(searchfor), case=False)]
df2

name  ...        date
0       ACER ASPIRE 3 A314 - INTEL CELERON N4000 - 14 ...  ...  2020-10-02
1       Laptop Acer Aspire 3 Celeron N4120 256GB SSD 4...  ...  2020-10-02
2       LAPTOP ACER ASPIRE 3 A314-21 A4 9120e RAM 4GB ...  ...  2020-10-02
3       Acer Aspire 3 A314 A4 9120 4GB 1TB W10 14.0 DV...  ...  2020-10-02
4       PC All In One acer Aspire C24-960 8GB 1TB Core...  ...  2020-10-02
...                                                   ...  ...         ...
150683  ASUS VIVOBOOK FLIP TM420IA RYZEN 5-4500U 8GB 5...  ...  2021-01-26
150684  Laptop Asus Vivobook K413FQ i5 8gb 512Gb 14fhd...  ...  2021-01-26
150685  Asus vivobook A416ma intel N4020 4gb 256gb 14 ...  ...  2021-01-26
150686  ASUS VIVOBOOK A407UF I5 8250U 4GB 256GB MX130 ...  ...  2021-01-26
150687  LAPTOP ASUS VIVOBOOK K413EQ I7-1165G7 8GB 512G...  ...  2021-01-26

[146515 rows x 5 columns]

In [13]:
# cek hasil buangan
filter = df2[df2['name'].str.contains('pc gaming', na=False)]
filter

Empty DataFrame
Columns: [name, price, location, sold, date]
Index: []

##Menguraikan Data
Untuk membuat data menjadi lebih spesifik, maka dataset perlu diuraikan menjadi beberapa kolom berdasarkan spesifikasi laptop. Informasi tentang spesifikasi laptop ini dapat dilihat dari kolom `name`.

Kolom baru yang akan dibentuk adalah :
* `type`
* `merk`
* `prosesor`
* `memory_size`
* `disk_size`
* `disk_type`

###Kolom **`type`**

In [14]:
# menduplikasi dataframe untuk mengantisipasi kekeliruan eksekusi
df3 = df2.copy(deep=True)
df3

name  ...        date
0       ACER ASPIRE 3 A314 - INTEL CELERON N4000 - 14 ...  ...  2020-10-02
1       Laptop Acer Aspire 3 Celeron N4120 256GB SSD 4...  ...  2020-10-02
2       LAPTOP ACER ASPIRE 3 A314-21 A4 9120e RAM 4GB ...  ...  2020-10-02
3       Acer Aspire 3 A314 A4 9120 4GB 1TB W10 14.0 DV...  ...  2020-10-02
4       PC All In One acer Aspire C24-960 8GB 1TB Core...  ...  2020-10-02
...                                                   ...  ...         ...
150683  ASUS VIVOBOOK FLIP TM420IA RYZEN 5-4500U 8GB 5...  ...  2021-01-26
150684  Laptop Asus Vivobook K413FQ i5 8gb 512Gb 14fhd...  ...  2021-01-26
150685  Asus vivobook A416ma intel N4020 4gb 256gb 14 ...  ...  2021-01-26
150686  ASUS VIVOBOOK A407UF I5 8250U 4GB 256GB MX130 ...  ...  2021-01-26
150687  LAPTOP ASUS VIVOBOOK K413EQ I7-1165G7 8GB 512G...  ...  2021-01-26

[146515 rows x 5 columns]

In [15]:
#Menambahkan kolom kategori laptop berdasarkan kolom name yang mengandung sebagian string

aspire = ['aspire','a314','z1402','a315','a514','es1','e5','1402']
rog = ['rog','g532']
tuf = ['tuf','fx505','fx506']
vivobook = ['vivobook','a407','a412','a409','a416','s430','m409','x441','x407','x456','x541','e402','k413','s333','a410']
inspiron = ['inspiron','inspirron']
latitude = ['latitude','latitudhe','3350','e5470','rugged','6410','6420','6430','6510','e4310','e5510','e6410','e7250']
pavilion = ['pavilion','pav']
hp_14 = ['hp 14','14s']
elitebook = ['elitebook','elitbook','elietbook','elite','hp 820','hp 8440','hp 8470','hp 8460','revolve']
probook = ['probook','hp 6450','hp pro']
ideapad = ['ideapad','slim','14ast','s145','l340']
thinkpad = ['thinkpad','x250','x240','x230','x201','carbon','l420','l430','l440','t410','t420','t430','t440','t510','t530','t460','k4250']
portege = ['portege','portage']
dynabook = ['dynabook','b551','b552','b553','r732','r734','r63']
lifebook = ['lifebook','a573','p771']

cond_list = [df3['name'].str.contains('|'.join(aspire), case=False),
             df3['name'].str.contains('nitro', case=False),
             df3['name'].str.contains('swift', case=False),
             df3['name'].str.contains('predator', case=False),
             df3['name'].str.contains('travelmate', case=False),
             df3['name'].str.contains('|'.join(rog), case=False),
             df3['name'].str.contains('|'.join(tuf), case=False),
             df3['name'].str.contains('|'.join(vivobook), case=False),
             df3['name'].str.contains('zenbook', case=False),
             df3['name'].str.contains('|'.join(inspiron), case=False),
             df3['name'].str.contains('alienware', case=False),
             df3['name'].str.contains('|'.join(latitude), case=False),
             df3['name'].str.contains('g5-5590', case=False),
             df3['name'].str.contains('optiplex', case=False),
             df3['name'].str.contains('precision', case=False),
             df3['name'].str.contains('vostro', case=False),
             df3['name'].str.contains('xps', case=False),
             df3['name'].str.contains('envy', case=False),
             df3['name'].str.contains('omen', case=False),
             df3['name'].str.contains('|'.join(pavilion), case=False),
             df3['name'].str.contains('hp 11', case=False),
             df3['name'].str.contains('|'.join(hp_14), case=False),
             df3['name'].str.contains('hp 15', case=False),
             df3['name'].str.contains('hp 240', case=False),
             df3['name'].str.contains('hp 245', case=False),
             df3['name'].str.contains('|'.join(elitebook), case=False),
             df3['name'].str.contains('spectre', case=False),
             df3['name'].str.contains('|'.join(probook), case=False),
             df3['name'].str.contains('|'.join(ideapad), case=False),
             df3['name'].str.contains('legion', case=False),
             df3['name'].str.contains('yoga', case=False),
             df3['name'].str.contains('|'.join(thinkpad), case=False),
             df3['name'].str.contains('k2450', case=False),
             df3['name'].str.contains('v14', case=False),
             df3['name'].str.contains('v15', case=False),
             df3['name'].str.contains('thinkbook', case=False),
             df3['name'].str.contains('alpha', case=False),
             df3['name'].str.contains('gf63', case=False),
             df3['name'].str.contains('gl65', case=False),
             df3['name'].str.contains('gf65', case=False),
             df3['name'].str.contains('modern', case=False),
             df3['name'].str.contains('prestige', case=False),
             df3['name'].str.contains('bravo', case=False),
             df3['name'].str.contains('blade', case=False),
             df3['name'].str.contains('redmibook', case=False),
             df3['name'].str.contains('mi gaming', case=False),
             df3['name'].str.contains('macbook air', case=False),
             df3['name'].str.contains('macbook pro', case=False),
             df3['name'].str.contains('|'.join(portege), case=False),
             df3['name'].str.contains('|'.join(dynabook), case=False),
             df3['name'].str.contains('|'.join(lifebook), case=False),
             df3['name'].str.contains('matebook', case=False),
             df3['name'].str.contains('gpd p2', case=False),
             df3['name'].str.contains('gpd win', case=False),
             df3['name'].str.contains('nec', case=False)]
choice_list = ['Aspire','Nitro','Swift','Predator','TravelMate','ROG','TUF','Vivobook','Zenbook','Inspiron','Alienware','Latitude','G5','Optiplex','Precision','Vostro','XPS','Envy','Omen','Pavilion','11','14','15','240','245','Elitebook','Spectre','Probook','Ideapad','Legion','Yoga','Thinkpad','K2450','V14','V15','Thinkbook','Alpha','GF63','GL65','GF65','Modern','Prestige','Bravo','Blade','Redmibook','Mi Gaming','Macbook Air','Macbook Pro','Portege','Dynabook','Lifebook','Matebook','GPD P2','GPD Win','Type Lainnya']

df3['type'] = np.select(cond_list, choice_list)
df3

name  ...      type
0       ACER ASPIRE 3 A314 - INTEL CELERON N4000 - 14 ...  ...    Aspire
1       Laptop Acer Aspire 3 Celeron N4120 256GB SSD 4...  ...    Aspire
2       LAPTOP ACER ASPIRE 3 A314-21 A4 9120e RAM 4GB ...  ...    Aspire
3       Acer Aspire 3 A314 A4 9120 4GB 1TB W10 14.0 DV...  ...    Aspire
4       PC All In One acer Aspire C24-960 8GB 1TB Core...  ...    Aspire
...                                                   ...  ...       ...
150683  ASUS VIVOBOOK FLIP TM420IA RYZEN 5-4500U 8GB 5...  ...  Vivobook
150684  Laptop Asus Vivobook K413FQ i5 8gb 512Gb 14fhd...  ...  Vivobook
150685  Asus vivobook A416ma intel N4020 4gb 256gb 14 ...  ...  Vivobook
150686  ASUS VIVOBOOK A407UF I5 8250U 4GB 256GB MX130 ...  ...  Vivobook
150687  LAPTOP ASUS VIVOBOOK K413EQ I7-1165G7 8GB 512G...  ...  Vivobook

[146515 rows x 6 columns]

In [16]:
# Memeriksa apakah dalam kolom ada 0
total_presence = len(df3[df3['type'] == '0'])
if(total_presence >= 0):
    print("Yes the null is present in the column. Total", total_presence)

Yes the null is present in the column. Total 163


In [17]:
# cek hasil filter
filter2 = df3[df3['type'] == '0']
filter2

name  ...  type
7629    LAPTOP HP MURAH AMD A6-7310 RAM 4 GB HARDDISK ...  ...     0
7707    LAPTOP HP MURAH AMD A6-7310 RAM 4 GB HARDDISK ...  ...     0
7785    LAPTOP HP MURAH AMD A6-7310 RAM 4 GB HARDDISK ...  ...     0
7859    LAPTOP HP MURAH AMD A6-7310 RAM 4 GB HARDDISK ...  ...     0
7937    LAPTOP HP MURAH AMD A6-7310 RAM 4 GB HARDDISK ...  ...     0
...                                                   ...  ...   ...
145208  EXTREME PC | i9-9900kf | RTX 2060 | 16GB | NVM...  ...     0
145327  EXTREME PC | i9-9900kf | RTX 2060 | 16GB | NVM...  ...     0
145329  EXTREME PC | i5-9600kf | RTX 2060 | 16GB | NVM...  ...     0
145476  EXTREME PC | i5-9600kf | RTX 2060 | 16GB | NVM...  ...     0
145506  EXTREME PC | i5-9600kf | RTX 2060 | 16GB | NVM...  ...     0

[163 rows x 6 columns]

Terdapat 163 data yang tidak memiliki informasi type di dalam kolom `name`. Juga masih terdapat data yang merupakan PC, bukan laptop seperti `EXTREME PC...`. Data ini perlu dibuang.

In [18]:
df4 = df3.drop(index=df3[df3['type'] == '0'].index)
df4

name  ...      type
0       ACER ASPIRE 3 A314 - INTEL CELERON N4000 - 14 ...  ...    Aspire
1       Laptop Acer Aspire 3 Celeron N4120 256GB SSD 4...  ...    Aspire
2       LAPTOP ACER ASPIRE 3 A314-21 A4 9120e RAM 4GB ...  ...    Aspire
3       Acer Aspire 3 A314 A4 9120 4GB 1TB W10 14.0 DV...  ...    Aspire
4       PC All In One acer Aspire C24-960 8GB 1TB Core...  ...    Aspire
...                                                   ...  ...       ...
150683  ASUS VIVOBOOK FLIP TM420IA RYZEN 5-4500U 8GB 5...  ...  Vivobook
150684  Laptop Asus Vivobook K413FQ i5 8gb 512Gb 14fhd...  ...  Vivobook
150685  Asus vivobook A416ma intel N4020 4gb 256gb 14 ...  ...  Vivobook
150686  ASUS VIVOBOOK A407UF I5 8250U 4GB 256GB MX130 ...  ...  Vivobook
150687  LAPTOP ASUS VIVOBOOK K413EQ I7-1165G7 8GB 512G...  ...  Vivobook

[146352 rows x 6 columns]

In [19]:
# cek hasil buangan
filter3 = df4[df4['type'] == '0']
filter3

Empty DataFrame
Columns: [name, price, location, sold, date, type]
Index: []

###Kolom **`merk`**

In [20]:
#Menambahkan kolom merk laptop berdasarkan kolom name yang mengandung sebagian string

acer = ['aspire','nitro','swift','predator','travelmate']
asus = ['rog','tuf','vivobook','zenbook']
dell = ['inspiron','alienware','latitude','g5','optiplex','precision','vostro','xps']
hp = ['envy','omen','pavilion','11','14','15','240','245','elitebook','spectre','probook']
lenovo = ['ideapad','legion','yoga','thinkpad','k2450','v14','v15','thinkbook']
msi = ['alpha','gf63','gl65','gf65','modern','prestige','bravo']
xiaomi = ['redmibook','mi gaming']
apple = ['macbook air','macbook pro']
toshiba = ['portege','dynabook']
gpd = ['gpd p2','gpd win']

cond_list2 = [df4['type'].str.contains('|'.join(acer), case=False),
             df4['type'].str.contains('|'.join(asus), case=False),
             df4['type'].str.contains('|'.join(dell), case=False),
             df4['type'].str.contains('|'.join(hp), case=False),
             df4['type'].str.contains('|'.join(lenovo), case=False),
             df4['type'].str.contains('|'.join(msi), case=False),
             df4['type'].str.contains('blade', case=False),
             df4['type'].str.contains('|'.join(xiaomi), case=False),
             df4['type'].str.contains('|'.join(apple), case=False),
             df4['type'].str.contains('|'.join(toshiba), case=False),
             df4['type'].str.contains('lifebook', case=False),
             df4['type'].str.contains('matebook', case=False),
             df4['type'].str.contains('|'.join(gpd), case=False),
             df4['type'].str.contains('type lainnya', case=False)]
choice_list2 = ['Acer', 'Asus', 'Dell', 'HP', 'Lenovo', 'MSI','Razer','Xiaomi','Apple','Toshiba','Fujitsu','Huawei','GPD','Merk Lainnya']

df4['merk'] = np.select(cond_list2, choice_list2)
df4

name  ...  merk
0       ACER ASPIRE 3 A314 - INTEL CELERON N4000 - 14 ...  ...  Acer
1       Laptop Acer Aspire 3 Celeron N4120 256GB SSD 4...  ...  Acer
2       LAPTOP ACER ASPIRE 3 A314-21 A4 9120e RAM 4GB ...  ...  Acer
3       Acer Aspire 3 A314 A4 9120 4GB 1TB W10 14.0 DV...  ...  Acer
4       PC All In One acer Aspire C24-960 8GB 1TB Core...  ...  Acer
...                                                   ...  ...   ...
150683  ASUS VIVOBOOK FLIP TM420IA RYZEN 5-4500U 8GB 5...  ...  Asus
150684  Laptop Asus Vivobook K413FQ i5 8gb 512Gb 14fhd...  ...  Asus
150685  Asus vivobook A416ma intel N4020 4gb 256gb 14 ...  ...  Asus
150686  ASUS VIVOBOOK A407UF I5 8250U 4GB 256GB MX130 ...  ...  Asus
150687  LAPTOP ASUS VIVOBOOK K413EQ I7-1165G7 8GB 512G...  ...  Asus

[146352 rows x 7 columns]

In [21]:
# Memeriksa apakah dalam kolom ada 0
total_presence = len(df4[df4['merk'] == '0'])
if(total_presence >= 0):
    print("Yes the null is present in the column. Total", total_presence)

Yes the null is present in the column. Total 0


In [22]:
# cek hasil pemeriksaan
filter3 = df4[df4['merk'] == '0']
filter3

Empty DataFrame
Columns: [name, price, location, sold, date, type, merk]
Index: []

In [ ]:
df4.to_csv('tokped_clean_data2.csv', encoding='utf-8', index=False)

###Kolom `prosesor`

In [33]:
# menduplikasi dataframe untuk mengantisipasi kekeliruan eksekusi
df5 = df4.copy(deep=True)
df5

name  ...  merk
0       ACER ASPIRE 3 A314 - INTEL CELERON N4000 - 14 ...  ...  Acer
1       Laptop Acer Aspire 3 Celeron N4120 256GB SSD 4...  ...  Acer
2       LAPTOP ACER ASPIRE 3 A314-21 A4 9120e RAM 4GB ...  ...  Acer
3       Acer Aspire 3 A314 A4 9120 4GB 1TB W10 14.0 DV...  ...  Acer
4       PC All In One acer Aspire C24-960 8GB 1TB Core...  ...  Acer
...                                                   ...  ...   ...
150683  ASUS VIVOBOOK FLIP TM420IA RYZEN 5-4500U 8GB 5...  ...  Asus
150684  Laptop Asus Vivobook K413FQ i5 8gb 512Gb 14fhd...  ...  Asus
150685  Asus vivobook A416ma intel N4020 4gb 256gb 14 ...  ...  Asus
150686  ASUS VIVOBOOK A407UF I5 8250U 4GB 256GB MX130 ...  ...  Asus
150687  LAPTOP ASUS VIVOBOOK K413EQ I7-1165G7 8GB 512G...  ...  Asus

[146352 rows x 7 columns]

In [34]:
#Menambahkan kolom prosesor laptop berdasarkan kolom name yang mengandung sebagian string

celeron = ['celeron','n4000','n4120','n4020','n4100','3867u','n3350','n3050','n3060','n3160','n4205u','n5205u']
pent_silver = ['pentium silver','n5000','n5030']
pent_gold = ['pentium gold','5405u','6405u','pentium n6405']
dual = ['dual core','dualcore']
i5 = ['i5','1035g1','9300h']
i7 = ['i7','9750h']
ry_3 = ['ryzen 3','ryzen3','r3','reyzen 3','rzyen 3']
ry_5 = ['ryzen 5','ryzen5','r5','reyzen 5','¢ 5','ryzen-3500u','rayzen 5','ryze 5','r4500','ryzen pro 5','rayzen5','ryzen 2500u']
ry_7 = ['ryzen 7','ryzen7','r7','ryzen 8','¢ 7','rayzen 7']
ry_9 = ['ryzen 9','ryzen9','r9']
a9 = ['a9','9425','amd a 9']
a4 = ['amd a4','a4']

cond_list3 = [df5['name'].str.contains('|'.join(celeron), case=False),
             df5['name'].str.contains('|'.join(pent_silver), case=False),
             df5['name'].str.contains('|'.join(pent_gold), case=False),
             df5['name'].str.contains('|'.join(dual), case=False),
             df5['name'].str.contains('|'.join(i5), case=False),
             df5['name'].str.contains('|'.join(i7), case=False),
             df5['name'].str.contains('i3', case=False),
             df5['name'].str.contains('i9', case=False),
             df5['name'].str.contains('|'.join(ry_3), case=False),
             df5['name'].str.contains('|'.join(ry_5), case=False),
             df5['name'].str.contains('|'.join(ry_7), case=False),
             df5['name'].str.contains('|'.join(ry_9), case=False),
             df5['name'].str.contains('|'.join(a9), case=False),
             df5['name'].str.contains('|'.join(a4), case=False),
             df5['name'].str.contains('athlon', case=False),
             df5['name'].str.contains('amd a12|a12', case=False),
             df5['name'].str.contains('amd a10|a10', case=False),
             df5['name'].str.contains('amd a8|a8', case=False),
             df5['name'].str.contains('amd a6|a6', case=False),
             df5['name'].str.contains('amd e2|and e2 9000', case=False),
             df5['name'].str.contains('amd 3020', case=False),
             df5['name'].str.contains('amd 3050', case=False),
             df5['name'].str.contains('core m3|intel m3', case=False),
             df5['name'].str.contains('core m5', case=False),
             df5['name'].str.contains('quad core', case=False)]
choice_list3 = ['Intel Celeron', 'Intel Pentium Silver','Intel Pentium Gold','Intel Dual Core','Intel Core i5','Intel Core i7','Intel Core i3','Intel Core i9','AMD Ryzen 3', 'AMD Ryzen 5', 'AMD Ryzen 7', 'AMD Ryzen 9','AMD A9','AMD A4','Intel Athlon','AMD A12','AMD A10','AMD A8','AMD A6','AMD E2','AMD 3020','AMD 3050','Intel Core M3','Intel Core M5','Intel Quad Core']

df5['prosesor'] = np.select(cond_list3,choice_list3)
df5

name  ...       prosesor
0       ACER ASPIRE 3 A314 - INTEL CELERON N4000 - 14 ...  ...  Intel Celeron
1       Laptop Acer Aspire 3 Celeron N4120 256GB SSD 4...  ...  Intel Celeron
2       LAPTOP ACER ASPIRE 3 A314-21 A4 9120e RAM 4GB ...  ...         AMD A4
3       Acer Aspire 3 A314 A4 9120 4GB 1TB W10 14.0 DV...  ...         AMD A4
4       PC All In One acer Aspire C24-960 8GB 1TB Core...  ...  Intel Core i3
...                                                   ...  ...            ...
150683  ASUS VIVOBOOK FLIP TM420IA RYZEN 5-4500U 8GB 5...  ...    AMD Ryzen 5
150684  Laptop Asus Vivobook K413FQ i5 8gb 512Gb 14fhd...  ...  Intel Core i5
150685  Asus vivobook A416ma intel N4020 4gb 256gb 14 ...  ...  Intel Celeron
150686  ASUS VIVOBOOK A407UF I5 8250U 4GB 256GB MX130 ...  ...  Intel Core i5
150687  LAPTOP ASUS VIVOBOOK K413EQ I7-1165G7 8GB 512G...  ...  Intel Core i7

[146352 rows x 8 columns]

In [35]:
# Memeriksa apakah dalam kolom ada 0
total_presence = len(df5[df5['prosesor'] == '0'])
if(total_presence >= 0):
    print("Yes the null is present in the column. Total", total_presence)

Yes the null is present in the column. Total 11088


In [36]:
# cek hasil pemeriksaan
filter3 = df5[df5['prosesor'] == '0']
filter3

name  ...  prosesor
19      Laptop Acer Swift 3 Infinity SF314-55G Red NX....  ...         0
31      Laptop Gaming Acer Nitro 7 AN715-51 NH.Q5HSN.0...  ...         0
38                         Acer Aspire 3 A315-42-R0GT Red  ...         0
69      Acer Aspire 5 A514-52G-59RA_2 [UN.HMLSN.001] +...  ...         0
70                        LAPTOP ACER ASPIRE A314-32-P4AS  ...         0
...                                                   ...  ...       ...
150567                             ASUS VIVOBOOK E203 4GB  ...         0
150581                              Asus VivoBook E203MAH  ...         0
150587                               Laptop asus vivobook  ...         0
150631           ASUS VivoBook 14 K413FA-EK301T/302T/303T  ...         0
150634  Laptop ASUS VivoBook E203MA Notebook RAM 2 GB ...  ...         0

[11088 rows x 8 columns]

Terdapat `11088` data yang tidak memiliki informasi prosesor di dalam kolom `name`. Juga masih terdapat data yang tidak mencantumkan `disk_size` dan `memory_size`. Data ini perlu dibuang.

In [37]:
df6 = df5.drop(index=df5[df5['prosesor'] == '0'].index)
df6

name  ...       prosesor
0       ACER ASPIRE 3 A314 - INTEL CELERON N4000 - 14 ...  ...  Intel Celeron
1       Laptop Acer Aspire 3 Celeron N4120 256GB SSD 4...  ...  Intel Celeron
2       LAPTOP ACER ASPIRE 3 A314-21 A4 9120e RAM 4GB ...  ...         AMD A4
3       Acer Aspire 3 A314 A4 9120 4GB 1TB W10 14.0 DV...  ...         AMD A4
4       PC All In One acer Aspire C24-960 8GB 1TB Core...  ...  Intel Core i3
...                                                   ...  ...            ...
150683  ASUS VIVOBOOK FLIP TM420IA RYZEN 5-4500U 8GB 5...  ...    AMD Ryzen 5
150684  Laptop Asus Vivobook K413FQ i5 8gb 512Gb 14fhd...  ...  Intel Core i5
150685  Asus vivobook A416ma intel N4020 4gb 256gb 14 ...  ...  Intel Celeron
150686  ASUS VIVOBOOK A407UF I5 8250U 4GB 256GB MX130 ...  ...  Intel Core i5
150687  LAPTOP ASUS VIVOBOOK K413EQ I7-1165G7 8GB 512G...  ...  Intel Core i7

[135264 rows x 8 columns]

In [38]:
filter3.to_csv('check_processor5.csv', encoding='utf-8', index=False)

In [52]:
df6.to_csv('tokped_clean_data3.csv', encoding='utf-8', index=False)

###Kolom `memory_size`

In [59]:
# menduplikasi dataframe untuk mengantisipasi kekeliruan eksekusi
df7 = df6.copy(deep=True)
df7

name  ...       prosesor
0       ACER ASPIRE 3 A314 - INTEL CELERON N4000 - 14 ...  ...  Intel Celeron
1       Laptop Acer Aspire 3 Celeron N4120 256GB SSD 4...  ...  Intel Celeron
2       LAPTOP ACER ASPIRE 3 A314-21 A4 9120e RAM 4GB ...  ...         AMD A4
3       Acer Aspire 3 A314 A4 9120 4GB 1TB W10 14.0 DV...  ...         AMD A4
4       PC All In One acer Aspire C24-960 8GB 1TB Core...  ...  Intel Core i3
...                                                   ...  ...            ...
150683  ASUS VIVOBOOK FLIP TM420IA RYZEN 5-4500U 8GB 5...  ...    AMD Ryzen 5
150684  Laptop Asus Vivobook K413FQ i5 8gb 512Gb 14fhd...  ...  Intel Core i5
150685  Asus vivobook A416ma intel N4020 4gb 256gb 14 ...  ...  Intel Celeron
150686  ASUS VIVOBOOK A407UF I5 8250U 4GB 256GB MX130 ...  ...  Intel Core i5
150687  LAPTOP ASUS VIVOBOOK K413EQ I7-1165G7 8GB 512G...  ...  Intel Core i7

[135264 rows x 8 columns]

In [60]:
#Menambahkan kolom memory_size laptop berdasarkan kolom name yang mengandung sebagian string

gb2 = ['ram 2gb','2gb 500gb','2gb/500gb','2gb- 500gb','2gb ram - 500gb','2gb / 500gb','2gb office','ram2gb']
gb4 = ['4gb','4 gb','ram 4','4ddr4','4g 512','4/1tb','4g 1tb','4 1tb','4/ssd','4/hdd','4g 512ssd','4/500','4g 1t','4-256','4-1t']
gb8 = ['8gb','8 gb','8ddr4','8g/1tb','8/512','8g ssd','8g-512','8-512','8-256','8-1t','8-uma-1t','ram 8','8g 1tb','8/256']
gb12 = ['12gb','12 gb']
gb16 = ['16gb','16 gb','16-512','16/1tb','16ram']
gb32 = ['32gb','32 gb/1 tb']

cond_list4 = [df7['name'].str.contains('|'.join(gb32), case=False),
              df7['name'].str.contains('20gb', case=False),
              df7['name'].str.contains('|'.join(gb16), case=False),
              df7['name'].str.contains('|'.join(gb12), case=False),
              df7['name'].str.contains('10gb', case=False),
              df7['name'].str.contains('|'.join(gb8), case=False),
              df7['name'].str.contains('6gb', case=False),
              df7['name'].str.contains('|'.join(gb4), case=False),
              df7['name'].str.contains('3gb', case=False),
              df7['name'].str.contains('|'.join(gb2), case=False)]
choice_list4 = ['32 GB','20 GB','16 GB','12 GB','10 GB','8 GB','6 GB','4 GB','3 GB','2 GB']

df7['memory_size'] = np.select(cond_list4,choice_list4)
df7

name  ...  memory_size
0       ACER ASPIRE 3 A314 - INTEL CELERON N4000 - 14 ...  ...            0
1       Laptop Acer Aspire 3 Celeron N4120 256GB SSD 4...  ...         6 GB
2       LAPTOP ACER ASPIRE 3 A314-21 A4 9120e RAM 4GB ...  ...         4 GB
3       Acer Aspire 3 A314 A4 9120 4GB 1TB W10 14.0 DV...  ...         4 GB
4       PC All In One acer Aspire C24-960 8GB 1TB Core...  ...         8 GB
...                                                   ...  ...          ...
150683  ASUS VIVOBOOK FLIP TM420IA RYZEN 5-4500U 8GB 5...  ...        12 GB
150684  Laptop Asus Vivobook K413FQ i5 8gb 512Gb 14fhd...  ...        12 GB
150685  Asus vivobook A416ma intel N4020 4gb 256gb 14 ...  ...         6 GB
150686  ASUS VIVOBOOK A407UF I5 8250U 4GB 256GB MX130 ...  ...         6 GB
150687  LAPTOP ASUS VIVOBOOK K413EQ I7-1165G7 8GB 512G...  ...        12 GB

[135264 rows x 9 columns]

In [61]:
# Memeriksa apakah dalam kolom ada 0
total_presence = len(df7[df7['memory_size'] == '0'])
if(total_presence >= 0):
    print("Yes the null is present in the column. Total", total_presence)

Yes the null is present in the column. Total 12035


In [62]:
# cek hasil pemeriksaan
filter4 = df7[df7['memory_size'] == '0']
filter4

name  ...  memory_size
0       ACER ASPIRE 3 A314 - INTEL CELERON N4000 - 14 ...  ...            0
25      Acer Aspire 3 A314-21-93TJ [AMD A9-9420] [NX.H...  ...            0
53                      ACER Aspire 3 A314-32 intel N4000  ...            0
77      ACER ASPIRE 3 A314 - INTEL CELERON N4000 - 14 ...  ...            0
99      Acer Aspire 3 A314-21-4644 [AMD A4-9120E] [NX....  ...            0
...                                                   ...  ...          ...
150590                               Asus Vivobook A407UA  ...            0
150642         Laptop Asus Vivobook With Intel i3 7th Gen  ...            0
150654  Laptop Gaming Desain ASUS Vivobook A442U Core ...  ...            0
150664                ASUS Vivobook A409FA- BV351T/BV352T  ...            0
150669     laptop Asus vivobook A412FA intel pentium gold  ...            0

[12035 rows x 9 columns]

In [63]:
filter4.to_csv('check_memory7.csv', encoding='utf-8', index=False)

Terdapat `12035` data yang tidak memiliki informasi ukuran memory/ram di dalam kolom `name`. Data ini perlu dibuang.

In [64]:
df8 = df7.drop(index=df7[df7['memory_size'] == '0'].index)
df8

name  ...  memory_size
1       Laptop Acer Aspire 3 Celeron N4120 256GB SSD 4...  ...         6 GB
2       LAPTOP ACER ASPIRE 3 A314-21 A4 9120e RAM 4GB ...  ...         4 GB
3       Acer Aspire 3 A314 A4 9120 4GB 1TB W10 14.0 DV...  ...         4 GB
4       PC All In One acer Aspire C24-960 8GB 1TB Core...  ...         8 GB
5       Laptop Acer Aspire 3 - A314-32 - INTEL N4000 D...  ...         4 GB
...                                                   ...  ...          ...
150683  ASUS VIVOBOOK FLIP TM420IA RYZEN 5-4500U 8GB 5...  ...        12 GB
150684  Laptop Asus Vivobook K413FQ i5 8gb 512Gb 14fhd...  ...        12 GB
150685  Asus vivobook A416ma intel N4020 4gb 256gb 14 ...  ...         6 GB
150686  ASUS VIVOBOOK A407UF I5 8250U 4GB 256GB MX130 ...  ...         6 GB
150687  LAPTOP ASUS VIVOBOOK K413EQ I7-1165G7 8GB 512G...  ...        12 GB

[123229 rows x 9 columns]

In [66]:
df8.to_csv('tokped_clean_data4.csv', encoding='utf-8', index=False)

### Kolom `disk_size`

In [72]:
df9 = pd.read_csv('tokped_clean_data4.csv')
df9

name  ...  memory_size
0       Laptop Acer Aspire 3 Celeron N4120 256GB SSD 4...  ...         6 GB
1       LAPTOP ACER ASPIRE 3 A314-21 A4 9120e RAM 4GB ...  ...         4 GB
2       Acer Aspire 3 A314 A4 9120 4GB 1TB W10 14.0 DV...  ...         4 GB
3       PC All In One acer Aspire C24-960 8GB 1TB Core...  ...         8 GB
4       Laptop Acer Aspire 3 - A314-32 - INTEL N4000 D...  ...         4 GB
...                                                   ...  ...          ...
123224  ASUS VIVOBOOK FLIP TM420IA RYZEN 5-4500U 8GB 5...  ...        12 GB
123225  Laptop Asus Vivobook K413FQ i5 8gb 512Gb 14fhd...  ...        12 GB
123226  Asus vivobook A416ma intel N4020 4gb 256gb 14 ...  ...         6 GB
123227  ASUS VIVOBOOK A407UF I5 8250U 4GB 256GB MX130 ...  ...         6 GB
123228  LAPTOP ASUS VIVOBOOK K413EQ I7-1165G7 8GB 512G...  ...        12 GB

[123229 rows x 9 columns]

In [73]:
#Menambahkan kolom disk_size laptop berdasarkan kolom name yang mengandung sebagian string

tb2 = ['2 tb','2tb','2tbssd']
tb1 = ['1tb','1000gb','1tb ssd','1 tb','8-uma-1t','itb','1tssd','1t ssd','8-1t','4-1t','hdd 1tera','4g 1t','hdd 1t','4gb/1t']
gb512 = ['512','512gb','512 ssd','ssd 512']
gb500 = ['500gb','500 gb','hdd 500','500ggb','4/500','hdd500','4gb/500']
gb320 = ['320 gb','320gb','hdd 320']
gb256 = ['256','256gb','256ssd']
gb128 = ['128 gb','128 ssd','128ssd','128gb','ssd 128','ssd128']
gb240 = ['240gb','ssd 240','8gb/240ssd']
gb180 = ['ssd 180','180 gb']
gb120 = ['120ssd','ssd 120']

cond_list5 = [df9['name'].str.contains('4tb', case=False),
              df9['name'].str.contains('|'.join(tb2), case=False),
              df9['name'].str.contains('|'.join(tb1), case=False),
              df9['name'].str.contains('hdd 750 gransi', case=False),
              df9['name'].str.contains('|'.join(gb512), case=False),
              df9['name'].str.contains('|'.join(gb500), case=False),
              df9['name'].str.contains('|'.join(gb320), case=False),
              df9['name'].str.contains('|'.join(gb256), case=False),
              df9['name'].str.contains('250gb', case=False),
              df9['name'].str.contains('|'.join(gb240), case=False),
              df9['name'].str.contains('|'.join(gb180), case=False),
              df9['name'].str.contains('|'.join(gb128), case=False),
              df9['name'].str.contains('|'.join(gb120), case=False)]
choice_list5 = ['4 TB','2 TB','1 TB','750 GB','512 GB','500 GB','320 GB','256 GB','250 GB','240 GB','180 GB', '128 GB','120 GB']

df9['disk_size'] = np.select(cond_list5,choice_list5)
df9

name  ...  disk_size
0       Laptop Acer Aspire 3 Celeron N4120 256GB SSD 4...  ...     256 GB
1       LAPTOP ACER ASPIRE 3 A314-21 A4 9120e RAM 4GB ...  ...       1 TB
2       Acer Aspire 3 A314 A4 9120 4GB 1TB W10 14.0 DV...  ...       1 TB
3       PC All In One acer Aspire C24-960 8GB 1TB Core...  ...       1 TB
4       Laptop Acer Aspire 3 - A314-32 - INTEL N4000 D...  ...       1 TB
...                                                   ...  ...        ...
123224  ASUS VIVOBOOK FLIP TM420IA RYZEN 5-4500U 8GB 5...  ...     512 GB
123225  Laptop Asus Vivobook K413FQ i5 8gb 512Gb 14fhd...  ...     512 GB
123226  Asus vivobook A416ma intel N4020 4gb 256gb 14 ...  ...     256 GB
123227  ASUS VIVOBOOK A407UF I5 8250U 4GB 256GB MX130 ...  ...     256 GB
123228  LAPTOP ASUS VIVOBOOK K413EQ I7-1165G7 8GB 512G...  ...     512 GB

[123229 rows x 10 columns]

In [74]:
# Memeriksa apakah dalam kolom ada 0
total_presence = len(df9[df9['disk_size'] == '0'])
if(total_presence >= 0):
    print("Yes the null is present in the column. Total", total_presence)

Yes the null is present in the column. Total 4170


In [75]:
# cek hasil pemeriksaan
filter5 = df9[df9['disk_size'] == '0']
filter5

name  ...  disk_size
165     Acer aspire E5-475G/Core i5-7200U/Ram 4G/Nvidi...  ...          0
384                 Acer Core i3 ram4gb Aspire 1830T slim  ...          0
457     ACER ASPIRE 3 A315 # REYZEN 5 3500u 8GB 15inch...  ...          0
495     ACER ASPIRE 3 A315 # REYZEN 5-2500u 8GB 15inch...  ...          0
546     Pc Acer Aspire XC600 Intel Core i5 - 3440 @3.2...  ...          0
...                                                   ...  ...        ...
123049                 asus vivobook tm420ia ryzen 7 16gb  ...          0
123164                 asus vivobook tm420ia ryzen 7 16gb  ...          0
123197  ASUS VIVOBOOK A442U / CORE I5-8250 / DUAL VGA ...  ...          0
123206  Asus VivoBook Ultra K413FQ - Core i5-10210U|8G...  ...          0
123211  Asus Vivobook E203Mah Intel Celeron N4000 Ram ...  ...          0

[4170 rows x 10 columns]

In [76]:
filter5.to_csv('check_disk4.csv', encoding='utf-8', index=False)

Terdapat `4170` data yang tidak memiliki informasi ukuran disk di dalam kolom `name`. Data ini perlu dibuang.

In [77]:
df10 = df9.drop(index=df9[df9['disk_size'] == '0'].index)
df10

name  ...  disk_size
0       Laptop Acer Aspire 3 Celeron N4120 256GB SSD 4...  ...     256 GB
1       LAPTOP ACER ASPIRE 3 A314-21 A4 9120e RAM 4GB ...  ...       1 TB
2       Acer Aspire 3 A314 A4 9120 4GB 1TB W10 14.0 DV...  ...       1 TB
3       PC All In One acer Aspire C24-960 8GB 1TB Core...  ...       1 TB
4       Laptop Acer Aspire 3 - A314-32 - INTEL N4000 D...  ...       1 TB
...                                                   ...  ...        ...
123224  ASUS VIVOBOOK FLIP TM420IA RYZEN 5-4500U 8GB 5...  ...     512 GB
123225  Laptop Asus Vivobook K413FQ i5 8gb 512Gb 14fhd...  ...     512 GB
123226  Asus vivobook A416ma intel N4020 4gb 256gb 14 ...  ...     256 GB
123227  ASUS VIVOBOOK A407UF I5 8250U 4GB 256GB MX130 ...  ...     256 GB
123228  LAPTOP ASUS VIVOBOOK K413EQ I7-1165G7 8GB 512G...  ...     512 GB

[119059 rows x 10 columns]

In [83]:
filterxx = df10[df10['type'] == '0']
filterxx

Empty DataFrame
Columns: [name, price, location, sold, date, type, merk, prosesor, memory_size, disk_size]
Index: []

In [78]:
df10.to_csv('tokped_clean_data5.csv', encoding='utf-8', index=False)